python/GangaLHCb/__init__.py

```diff
@@ -82,16 +64,22 @@ def _store_root_version():
 
 def _store_dirac_environment():
     from GangaDirac.Lib.Utilities.DiracUtilities import write_env_cache, get_env
-    diracversion = _guess_version('LHCBDIRAC')
     platform = os.environ['CMTOPT']
+    diracversion = Ganga.Utility.Config.getConfig('LHCb')['DiracVersion']
     fdir = os.path.join(os.path.expanduser("~/.cache/Ganga/GangaLHCb"), platform)
     fname = os.path.join(fdir, diracversion)
     if not os.path.exists(fname) or not os.path.getsize(fname):
-        logger.info("Storing new LHCbDirac environment (%s:%s)" % (str(diracversion), str(platform)))
-        cmd = 'lb-run LHCBDIRAC {version} python -c "import os; print(dict(os.environ))"'.format(version=diracversion)
-        env = execute(cmd)  # grab the stdout text
-        env = eval(env)  # env is a string so convert it to a dict
+      cmd =  'lb-run LHCBDIRAC {version} python -c "import os; print(dict(os.environ))"'.format(version=diracversion)
+      env = execute(cmd)
+      try:
+        env = eval(env)
         write_env_cache(env, fname)
+        logger.info("Storing new LHCbDirac environment (%s:%s)" % (str(diracversion), str(platform)))
+      except SyntaxError:
```

Comment:
An `exit` here is putting the responsibility in the wrong place. The plugin system should run each plugin-load in a `try`/`except` statement, catch any thrown exceptions and either exit (if the plugin is essential) or continue (if the plugin can just be disabled). It is not up to the plugin to decide that the whole of Ganga must shutdown because it failed to load.

To put it another way, there's a difference between library and application code. Library code (e.g. the LHCb plugin) should never call things like `exit` because you never know how the application (the Ganga runtime) is using it.